File: arena-backend/src/entities/user.entity.ts
import {
  Column,
  CreateDateColumn,
  DeleteDateColumn,
  Entity,
  Index,
  JoinColumn,
  ManyToOne,
  PrimaryColumn,
  UpdateDateColumn,
} from "typeorm";
import { FileEntity } from "./file.entity";

@Entity({ name: "users" })
export class UserEntity {
  @PrimaryColumn({ type: "text" })
  userId: string;

  @Column({ type: "text", unique: true })
  uid: string;

  @Column({ type: "varchar", length: 8, unique: true })
  utag: string;

  @Index("idx_users_nick")
  @Column({ type: "varchar", length: 32 })
  nick: string;

  @Column({ type: "varchar", length: 255, nullable: true })
  email: string | null;

  @Column({ type: "varchar", length: 140, nullable: true })
  statusMessage: string | null;

  @Column({ type: "text", nullable: true })
  avatarFileId: string | null;

  @ManyToOne(() => FileEntity, { nullable: true })
  @JoinColumn({ name: "avatarFileId", referencedColumnName: "fileId" })
  avatar: FileEntity | null;

  @DeleteDateColumn({ type: "timestamptz", nullable: true })
  deletedAt: Date | null;

  @CreateDateColumn({ type: "timestamptz" })
  createdAt: Date;

  @UpdateDateColumn({ type: "timestamptz" })
  updatedAt: Date;
}

File: legacy/backend/src/main/java/app/sandori/arena/api/domain/user/UserEntity.java
package app.sandori.arena.api.domain.user;

import app.sandori.arena.api.domain.file.FileEntity;
import app.sandori.arena.api.global.entity.BaseTimeEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "users", indexes = {
        @Index(name = "idx_users_nick", columnList = "nick")
})
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class UserEntity extends BaseTimeEntity {

    @Id
    private String userId;

    @Column(nullable = false, unique = true)
    private String uid;

    @Column(nullable = false, unique = true, length = 8)
    private String utag;

    @Column(nullable = false, length = 32)
    private String nick;

    @Column(length = 255)
    private String email;

    @Column(length = 140)
    private String statusMessage;

    @Column
    private String avatarFileId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "avatarFileId", referencedColumnName = "fileId", insertable = false, updatable = false)
    private FileEntity avatar;

    @Builder
    public UserEntity(String userId, String uid, String utag, String nick,
                      String email, String statusMessage, String avatarFileId) {
        this.userId = userId;
        this.uid = uid;
        this.utag = utag;
        this.nick = nick;
        this.email = email;
        this.statusMessage = statusMessage;
        this.avatarFileId = avatarFileId;
    }
}
